src/test/java/seedu/ccacommander/logic/parser/UnenrolCommandParserTest.java

```diff
@@ -0,0 +1,142 @@
+import static seedu.ccacommander.logic.commands.CommandTestUtil.PREAMBLE_NON_EMPTY;
+import static seedu.ccacommander.logic.commands.CommandTestUtil.PREAMBLE_WHITESPACE;
+import static seedu.ccacommander.logic.commands.CommandTestUtil.VALID_EVENT_INDEX_TWO;
+import static seedu.ccacommander.logic.commands.CommandTestUtil.VALID_INDEX_ONE;
+import static seedu.ccacommander.logic.commands.CommandTestUtil.VALID_INDEX_TWO;
+import static seedu.ccacommander.logic.commands.CommandTestUtil.VALID_MEMBER_INDEX_ONE;
+import static seedu.ccacommander.logic.parser.CliSyntax.PREFIX_EVENT;
+import static seedu.ccacommander.logic.parser.CliSyntax.PREFIX_MEMBER;
+import static seedu.ccacommander.logic.parser.CommandParserTestUtil.assertParseFailure;
+import static seedu.ccacommander.logic.parser.CommandParserTestUtil.assertParseSuccess;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.ccacommander.logic.Messages;
+import seedu.ccacommander.logic.commands.UnenrolCommand;
+import seedu.ccacommander.logic.parser.exceptions.HandledParseException;
+import seedu.ccacommander.logic.parser.exceptions.ParseException;
+
+public class UnenrolCommandParserTest {
+    private UnenrolCommandParser parser = new UnenrolCommandParser();
+
+    @Test
+    public void parse_allFieldsPresent_success() {
+        UnenrolCommand expectedUnenrolEventCommand =
+                new UnenrolCommand(VALID_INDEX_ONE, VALID_INDEX_TWO);
+
+        // whitespace only preamble
+        assertParseSuccess(parser,
+                PREAMBLE_WHITESPACE + MEMBER_INDEX_DESC_ONE
+                        + EVENT_INDEX_DESC_TWO, expectedUnenrolEventCommand);
+    }
+
+    @Test
+    public void parse_repeatedValue_failure() {
+        String validExpectedEnrolmentString = MEMBER_INDEX_DESC_ONE + EVENT_INDEX_DESC_TWO;
+
+        // multiple member indexes
+        assertParseFailure(parser, MEMBER_INDEX_DESC_ONE + validExpectedEnrolmentString,
+                Messages.getErrorMessageForDuplicatePrefixes(PREFIX_MEMBER));
+
+        // multiple event indexes
+        assertParseFailure(parser, EVENT_INDEX_DESC_TWO + validExpectedEnrolmentString,
+                Messages.getErrorMessageForDuplicatePrefixes(PREFIX_EVENT));
+
+        // multiple fields repeated
+        assertParseFailure(parser,
+                validExpectedEnrolmentString + MEMBER_INDEX_DESC_ONE
+                        + EVENT_INDEX_DESC_TWO + validExpectedEnrolmentString,
+                Messages.getErrorMessageForDuplicatePrefixes(PREFIX_MEMBER, PREFIX_EVENT));
+
+        // invalid value followed by valid value
+
+        // invalid member index
+        assertParseFailure(parser, INVALID_MEMBER_INDEX_DESC + validExpectedEnrolmentString,
+                Messages.getErrorMessageForDuplicatePrefixes(PREFIX_MEMBER));
+
+        // invalid event index
+        assertParseFailure(parser, INVALID_EVENT_INDEX_DESC + validExpectedEnrolmentString,
+                Messages.getErrorMessageForDuplicatePrefixes(PREFIX_EVENT));
```

Comment:
Aren't these repeated values also since we have an invalid member then a valid member and a valid event?

Reply:
Makes sense, will make the change!